lib/ueberauth/strategy/facebook.ex

```diff
@@ -16,16 +16,13 @@ defmodule Ueberauth.Strategy.Facebook do
   Handles initial request for Facebook authentication.
   """
   def handle_request!(conn) do
-    auth_type = option(conn.params["auth_type"], conn, :auth_type)
-    scopes = option(conn.params["scope"], conn, :default_scope)
-
-    opts = [auth_type: auth_type, scope: scopes]
-
-    if conn.params["state"] do
-      opts = Keyword.put(opts, :state, conn.params["state"])
-    end
+    params = conn.params
+    |> maybe_replace_param(conn, "auth_type", :auth_type)
+    |> maybe_replace_param(conn, "scope", :default_scope)
 
+    opts = Enum.into(params, [])
```

Comment:
That wouldn't be cool because the params are coming from the url and could be anything.